hpx/runtime/threads/policies/thread_queue.hpp

```diff
@@ -416,27 +416,28 @@ namespace hpx { namespace threads { namespace policies {
                     thread_id_type tid(todelete);
                     --terminated_items_count_;
 
-                    thread_map_type::iterator it = thread_map_.find(tid);
-
                     // this thread has to be in this map
-                    HPX_ASSERT(it != thread_map_.end());
-
-                    recycle_thread(*it);
+                    HPX_ASSERT(thread_map_.find(tid) != thread_map_.end());
 
-                    thread_map_.erase(it);
+                    bool deleted = thread_map_.erase(tid) != 0;
+                    HPX_ASSERT(deleted);
+                    (void)deleted;
```

Comment:
```suggestion
                    (HPX_UNUSED(deleted);
```

The changes in this function aren't intended to change behaviour its behaviour, right? It's not obvious to me if there's some functional change from recycling the thread last (other than allowing you to check that the `erase` is successful).

Reply:
This change was done to make it consistent with another spot in the code where we do similar things. No functional change was intended.